quantum/process_keycode/process_autocorrect.c

```diff
@@ -0,0 +1,201 @@
+        case QK_SWAP_HANDS ... QK_SWAP_HANDS_MAX:
+            if (*keycode >= 0x56F0 || !record->tap.count) {
+                return false;
+            }
+            *keycode &= 0xFF;
+            break;
+#endif
+#ifndef NO_ACTION_ONESHOT
+        case QK_ONE_SHOT_MOD ... QK_ONE_SHOT_MOD_MAX:
+            if ((*keycode & 0xF) == MOD_LSFT) {
+                return false;
+            }
+            // fall through intentionally
+#endif
+        case QK_LSFT ... QK_LSFT + 255:
+        case QK_RSFT ... QK_RSFT + 255:
+            *keycode &= 0xFF; // Get the basic keycode.
+            break;
+        default:
+            // Disable autocorrect while a mod other than shift is active.
+            if ((mods & ~MOD_MASK_SHIFT) != 0) {
+                typo_buffer_size = 0;
+                return false;
+            }
+    }
+    return true;
+}
+
+/**
+ * @brief Process handler for autocorrect feature
+ *
+ * @param keycode Keycode registered by matrix press, per keymap
+ * @param record keyrecord_t structure
+ * @return true Continue processing keycodes, and send to host
+ * @return false Stop processing keycodes, and don't send to host
+ */
+
+bool process_autocorrect(uint16_t keycode, keyrecord_t *record) {
+    static uint8_t typo_buffer[AUTOCORRECT_MAX_LENGTH] = {KC_SPC};
+    static uint8_t typo_buffer_size                    = 1;
+#ifndef NO_ACTION_ONESHOT
+    const uint8_t mods = get_mods() | get_oneshot_mods();
+#else
+    const uint8_t mods = get_mods();
+#endif // NO_ACTION_ONESHOT
+
+    if (keycode == AUTOCRT) {
+        if (record->event.pressed) {
+            typo_buffer_size = 0;
+            keymap_config.autocorrect_enable ^= 1;
+            eeconfig_update_keymap(keymap_config.raw);
+        }
+        return false;
+    }
+
+    if (!keymap_config.autocorrect_enable) {
+        typo_buffer_size = 0;
+        return true;
+    }
+
+    if (!record->event.pressed) {
+        return true;
+    }
+
+    if (!process_autocorrect_user(&keycode, record, &typo_buffer_size, mods)) {
+        return true;
+    }
+    // Subtract buffer for Backspace key, reset for other non-alpha.
+    if (keycode == KC_QUOT) {
+        // Treat " (shifted ') as a word boundary.
+        if ((mods & MOD_MASK_SHIFT) != 0) {
+            keycode = KC_SPC;
+        }
+    } else if (!(KC_A <= keycode && keycode <= KC_Z)) {
+        if (keycode == KC_BSPC) {
+            // Remove last character from the buffer.
+            if (typo_buffer_size > 0) {
+                --typo_buffer_size;
+            }
+            return true;
+        } else if (KC_1 <= keycode && keycode <= KC_SLSH && keycode != KC_ESC) {
+            // Set a word boundary if space, period, digit, etc. is pressed.
+            // Behave more conservatively for the enter key. Reset, so that enter
+            // can't be used on a word ending.
+            if (keycode == KC_ENT || (keycode == KC_MINUS && (get_mods() | get_oneshot_mods()) & MOD_MASK_SHIFT)) {
```

Comment:
```suggestion
    } else if (!(KC_A <= (uint8_t)keycode && (uint8_t)keycode <= KC_Z)) {
        if ((uint8_t)keycode == KC_BSPC) {
            // Remove last character from the buffer.
            if (typo_buffer_size > 0) {
                --typo_buffer_size;
            }
            return true;
        } else if (KC_1 <= (uint8_t)keycode && (uint8_t)keycode <= KC_SLSH && (uint8_t)keycode != KC_ESC) {
            // Set a word boundary if space, period, digit, etc. is pressed.
            // Behave more conservatively for the enter key. Reset, so that enter
            // can't be used on a word ending.
            if ((uint8_t)keycode == KC_ENT || ((uint8_t)keycode == KC_MINUS && (get_mods() | get_oneshot_mods()) & MOD_MASK_SHIFT)) {
```